Closes #204 - Add WebSerial support.

patrykwegrzyn opened the pull request:
Hi @colinbdclark i'm not familiar with testing framework you are using so this may need some work on your end.
Please review and let me know if is anything else i can do to help.
Best regards 
Pat

colinbdclark commented:
Hi @patrykwegrzyn, thanks so much for this PR. I'm swamped at the moment but will definitely take a close look and review this as soon as I can.

It looks like your editor maybe has automatically reformatted a number of files that you probably didn't mean to change. Can you make sure any white space changes etc. are omitted from your PR so that it's easier to review what has actually changed? Thanks so much!

patrykwegrzyn commented:
Hi @colinbdclark  can  you let me know what settings you are using ? im using prettier with standard settings i may be able to and do it but it may be painful.
As alternative I can  make comments for you to highlight to what i  have changed, i only added 1 line to Gruntfile , modified browser test and added osc-web-serialport.js . I did run build tho , probably that's why it looks like a lot stuff was changed.

patrykwegrzyn commented on Gruntfile.js:
```diff
@@ -9,135 +9,142 @@
+    oscWeb: [
+      "src/osc-transports.js",
+      "src/platforms/osc-websocket-client.js",
+      "src/platforms/osc-web-serialport.js",
```
I only changed this in a Grunt file

patrykwegrzyn commented on src/platforms/osc-web-serialport.js:
```diff
@@ -0,0 +1,110 @@
+ */
+
+/*global WebSerial, require*/
+var osc = osc || require("../osc.js");
```
Followed your template and added new file to enable WebSerial support

patrykwegrzyn commented on tests/osc-web-tests.js:
```diff
@@ -10,12 +10,12 @@
-}());
+  QUnit.test("Serial port support is not loaded", function () {
+    QUnit.expect(1);
+    QUnit.ok(osc.supportsSerial);
```
I modified this line to get tests to pass.

ahfontaine commented:
Hi, just checking if this PR will ever get accepted? I am getting a lot of Webpack 5 issues due to polyfill not being supported anymore. I can revert but, would love to upgrade my project

colinbdclark commented:
Hi @ahfontaine, I am in the midst of a big career transition that has been occupying my time recently, but the good news is that I anticipate having more time to focus on my open source projects soon. I'm sorry for the delay with this, and do have the intention of reviewing and merging when life permits.

coder0107git commented:
@colinbdclark Any updates on this?
